src/App/component/PullRequestsPage.jsx

```diff
@@ -0,0 +1,170 @@
+  },
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: '#fff',
+  },
+}))
+
+function PullRequestsPage(prop) {
+  const classes = useStyles()
+  const [pullRequestListData, setPullRequestListData] = useState([])
+  const [dataForPullRequestChart, setDataForPullRequestChart] = useState({ labels: [], data: { opened: [], closed: [], merged: [] } })
+
+  const [currentProject, setCurrentProject] = useState({})
+
+  const projectId = localStorage.getItem("projectId")
+  const jwtToken = localStorage.getItem("jwtToken")
+
+  const [open, setOpen] = useState(false);
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const handleToggle = () => {
+    setOpen(!open);
+  };
+
+  useEffect(() => {
+    Axios.get(`http://localhost:9100/pvs-api/project/1/${projectId}`,
+      { headers: { "Authorization": `${jwtToken}` } })
+      .then((response) => {
+        setCurrentProject(response.data)
+      })
+      .catch((e) => {
+        alert(e.response?.status)
+        console.error(e)
+      })
+  }, [])
+
+  const getPullRequestsFromGitHub = () => {
+    const githubRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'github')
+    if (githubRepo !== undefined) {
+      const query = githubRepo.url.split("github.com/")[1]
+
+      // todo need reafctor with async
+      Axios.get(`http://localhost:9100/pvs-api/github/pullRequests/${query}`,
+        { headers: { "Authorization": `${jwtToken}` } })
+        .then((response) => {
+          if (response?.data) {
+            setPullRequestListData(response.data)
+          }
+        })
+        .catch((e) => {
+          alert(e.response?.status);
+          console.error(e)
+        })
+    }
+  }
+
+  useEffect(() => {
+    if (Object.keys(currentProject).length !== 0) {
+      handleToggle()
+      getPullRequestsFromGitHub()
+      handleClose()
+    }
+  }, [currentProject, prop.startMonth, prop.endMonth])
+
+  // Generate the pull-request chart
+  useEffect(() => {
+    const { startMonth, endMonth } = prop
+    let chartDataset = { labels: [], data: { merged: [], closed: [], opened: [] } }
+    let pullRequestListDataSortedByCreatedAt = pullRequestListData
+    let pullRequestListDataSortedByClosedAt = pullRequestListData
+    let pullRequestListDataSortedByMergedAt = pullRequestListData
+
+    // Sort data by date
+    if (pullRequestListData !== undefined) {
+      [].slice.call(pullRequestListDataSortedByCreatedAt).sort((a, b) => a.createdAt - b.createdAt);
+      [].slice.call(pullRequestListDataSortedByClosedAt).sort((a, b) => a.closedAt - b.closedAt);
+      [].slice.call(pullRequestListDataSortedByMergedAt).sort((a, b) => a.mergedAt - b.mergedAt);
+    }
+
+    if (pullRequestListDataSortedByCreatedAt.length > 0) {
+      // Calculate the number of pull requests for each month in the selected range
+      for (let month = moment(startMonth); month < moment(endMonth).add(1, 'months'); month = month.add(1, 'months')) {
+        let index

```

Comment:
Suggest rename this variable.
It may be confused for what is a `index` here, especially inside a for-loop.

For example:
```js
for (let i = 0; i < 5; i++) {
  let j; // <== WHAT is THIS ??????
}